compiler/passes/codegen.cpp

```diff
@@ -1489,9 +1489,12 @@ void codegen(void) {
       USR_WARN("C code generation for packed pointers not supported");
   }
 
-  if(fIncrementalCompilation && fFastFlag)
-    USR_WARN("Compiling with --incremental and --fast together can lead to "
-             "slower execution time than compiling with --fast only");
+  std::size_t optimizationsEnabled = ccflags.find("-O");
+  if(fIncrementalCompilation && ( fFastFlag||
+      optimizationsEnabled!=std::string::npos ))
```

Comment:
Oh wow, this is more in-depth than I was asking for (which suggests I may not have been entirely clear in my original feedback).  If asked, I was going to say that it would be OK to treat "--ccflags" as being a manual override, outside of the space that we should be concerned about generating warnings for (particularly since there may be any number of optimization flags that a user may supply to the back-end compiler which may or may not start with -O and therefore could be hard to detect).

The case I was really worried about was the 'chpl' -O/--optimize flag because it's our portable way of saying "turn on back-end C compiler options" and it is enabled by the --fast flag.  In speaking about -O3, I was thinking only of the -O3 (or whatever) that our '-O/--optimize' implied (so may have been unclear there), not necessarily user-supplied -O3 flags (not that I object to trying for that as well).

In any case, the 'chpl -O --incremental' case still does not warn which is what prompted my original comment.  I think the fix is simply to replace the check against fFastFlag with a check against optimizeCCode, which will also warn for the --fast case since --fast implies -O.

Sorry to have been unclear.